src/components/admin/InviteAdminModal/InviteAdminModal.tsx

```diff
@@ -27,29 +43,84 @@ export const InviteAdminModal: React.FC<InviteAdminModalProps> = ({
     []
   );
 
-  const { register } = useForm({
+  const { register, watch } = useForm({
     reValidateMode: "onChange",
     defaultValues,
   });
 
-  const addAdmin = async () => {
+  const searchQuery = watch("userName");
+
+  const { value: algoliaValue } = useAlgoliaSearch(searchQuery);
```

Comment:
[Q] Is the hook throttling too many changes at too short intervals?

Reply:
Honestly, I didn't notice any throttle/debounce behavior whatsoever. I think we need to check the hook and improve it.

Reply:
K, how about this?
```suggestion
  // @debt check if searchQuery is triggering too many algoliaValue changes per unit of time
  const searchQuery = watch("userName");
  const { value: algoliaValue } = useAlgoliaSearch(searchQuery);
```